opensearch/src/test/java/org/opensearch/sql/opensearch/storage/script/filter/FilterQueryBuilderTest.java

```diff
@@ -634,6 +636,156 @@ void should_build_match_phrase_query_with_custom_parameters() {
+        StringUtils.format("Actual %s doesn't match neither expected %s", actual, expected));
+
+    expected = "{\n"
+        + "  \"wildcard\" : {\n"
+        + "    \"field\" : {\n"
+        + "      \"wildcard\" : \"search query?\",\n"
+        + "      \"boost\" : 1.0\n"
+        + "    }\n"
+        + "  }\n"
+        + "}";
+    actual = buildQuery(dsl.wildcard_query(
+        dsl.namedArgument("field", literal("field")),
+        dsl.namedArgument("query", literal("search query_"))));
+
+    assertTrue(new JSONObject(expected).similar(new JSONObject(actual)),
+        StringUtils.format("Actual %s doesn't match neither expected %s", actual, expected));
+  }
+
+  @Test
+  void wildcard_query_escape_sql_wildcard_characters() {
+    var expected = "{\n"
+        + "  \"wildcard\" : {\n"
+        + "    \"field\" : {\n"
+        + "      \"wildcard\" : \"search query%\",\n"
+        + "      \"boost\" : 1.0\n"
+        + "    }\n"
+        + "  }\n"
+        + "}";
+    var actual = buildQuery(dsl.wildcard_query(
+        dsl.namedArgument("field", literal("field")),
+        dsl.namedArgument("query", literal("search query\\%"))));
+
+    assertTrue(new JSONObject(expected).similar(new JSONObject(actual)),
+        StringUtils.format("Actual %s doesn't match neither expected %s", actual, expected));
+
+    expected = "{\n"
+        + "  \"wildcard\" : {\n"
+        + "    \"field\" : {\n"
+        + "      \"wildcard\" : \"search query_\",\n"
+        + "      \"boost\" : 1.0\n"
+        + "    }\n"
+        + "  }\n"
+        + "}";
+    actual = buildQuery(dsl.wildcard_query(
+        dsl.namedArgument("field", literal("field")),
+        dsl.namedArgument("query", literal("search query\\_"))));
+
+    assertTrue(new JSONObject(expected).similar(new JSONObject(actual)),
+        StringUtils.format("Actual %s doesn't match neither expected %s", actual, expected));
+
+    expected = "{\n"
+        + "  \"wildcard\" : {\n"
+        + "    \"field\" : {\n"
+        + "      \"wildcard\" : \"search query\\\\*\",\n"
+        + "      \"boost\" : 1.0\n"
+        + "    }\n"
+        + "  }\n"
+        + "}";
+    actual = buildQuery(dsl.wildcard_query(
+        dsl.namedArgument("field", literal("field")),
+        dsl.namedArgument("query", literal("search query\\*"))));
+
+    assertTrue(new JSONObject(expected).similar(new JSONObject(actual)),
+        StringUtils.format("Actual %s doesn't match neither expected %s", actual, expected));
+
+    expected = "{\n"
+        + "  \"wildcard\" : {\n"
+        + "    \"field\" : {\n"
+        + "      \"wildcard\" : \"search query\\\\?\",\n"
+        + "      \"boost\" : 1.0\n"
+        + "    }\n"
+        + "  }\n"
+        + "}";
+    actual = buildQuery(dsl.wildcard_query(
+        dsl.namedArgument("field", literal("field")),
+        dsl.namedArgument("query", literal("search query\\?"))));
+
+    assertTrue(new JSONObject(expected).similar(new JSONObject(actual)),
```

Comment:
How come you are not using assertJsonEquals in these tests?

Reply:
Good catch. I copied it from one of the other tests but this was being used because it had multiple comparisons. I will update this to be assertJsonEquals. Thanks!

Reply:
Changed in https://github.com/Bit-Quill/opensearch-project-sql/commit/a2ca906329ab194b845b8078eafeae48eb4bf1c9